generated-docs/FRP/Event.md

```diff
@@ -54,49 +63,84 @@ IsEvent Event
 never :: forall a. Event a
 ```
 
-#### `filter`
+#### `keepLatest`
 
 ``` purescript
-filter :: forall a. (a -> Boolean) -> Event a -> Event a
+keepLatest :: forall a. Event (Event a) -> Event a
 ```
 
-Create an `Event` which only fires when a predicate holds.
+Flatten a nested `Event`, reporting values only from the most recent
+inner `Event`.
 
 #### `subscribe`
 
 ``` purescript
-subscribe :: forall eff a r. Event a -> (a -> Eff (frp :: FRP | eff) r) -> Eff (frp :: FRP | eff) Unit
+subscribe :: forall eff a r. Event a -> (a -> Eff (frp :: FRP | eff) r) -> Eff (frp :: FRP | eff) (Eff (frp :: FRP | eff) Unit)
 ```
 
 Subscribe to an `Event` by providing a callback.
 
-#### `create`
+`subscribe` returns a canceller function.
+
+
+### Re-exported from Data.Filterable:
+
+#### `Filterable`
 
 ``` purescript
-create :: forall eff a. Eff (frp :: FRP | eff) { event :: Event a, push :: a -> Eff (frp :: FRP | eff) Unit }
+class (Functor f) <= Filterable f  where
+  filterMap :: forall a b. (a -> Maybe b) -> f a -> f b
 ```
 
-Create an event and a function which supplies a value to that event.
+`Filterable` represents data structures which can be _partitioned_/_filtered_.
 
+- `partitionMap` - partition a data structure based on an either predicate.
+- `partition` - partition a data structure based on boolean predicate.
+- `filterMap` - map over a data structure and filter based on a maybe.
+- `filter` - filter a data structure based on a boolean.
+
+Laws:
+- `map f ≡ filterMap (Just <<< f)`
+- `filter ≡ filterMap <<< maybeBool`
+- `filterMap p ≡ filter (isJust <<< p)`
```

Comment:
Might be overkill, but you can absolutely write similar laws for `partition`, e.g.

```
map f === map _.right (partitionMap (Right <<< f))
```

I get that this is probably more useful to @LiamGoodacre, as I think these laws came from the `Filterable` docs.

Reply:
(Will accept PR for updated laws.)

Reply:
I think I PR'd a while ago with some default implementations - I'll PR to reflect them tonight/this weekend :)